tests/sqlparser_snowflake.rs

```diff
@@ -99,6 +99,12 @@ fn test_single_table_in_parenthesis() {
     );
 }
 
+#[test]
+fn test_snowflake_variables() {
```

Comment:
I think it would be good to have a test that the parsed structure is actually what is expected -- for example https://github.com/sqlparser-rs/sqlparser-rs/blob/c884fbc/tests/sqlparser_common.rs#L138-L168